### PR TITLE
dev-extbase-fluid: Some fixes to OAI-PMH interface

### PR DESCRIPTION
**Pull request by dvoracek-slub**
This fixes some issues that I noticed while writing tests (#756).

---

- Fix listing records and identifiers (set $resultSet['elements'])
- Fix XML error when embedding METS documents (there has been an excess angle bracket)
- Don't include <resumptionToken> tag when there's no token
- Fix value of cursor attribute in <resumptionToken>
- Don't show request parameters when there's a badVerb or badArgument
- Restore fallback "earliestDatestamp" when no document is found

### CODE REVIEW

File: Classes/Controller/OaiPmhController.php [hunk 1]
```diff
@@ -240,10 +240,7 @@ protected function getMetsData(array $record)
             // Get root element.
             $root = $xml->getElementsByTagNameNS($this->formats['mets']['namespace'], 'mets');
             if ($root->item(0) instanceof \DOMNode) {
-                // Import node into \DOMDocument.
-                $mets = $xml->saveXML();
-                // Remove leading line
-                $mets = substr($mets, strpos($mets, '>'));
+                $mets = $xml->saveXML($root->item(0));
```
Comment (albig) [inline Classes/Controller/OaiPmhController.php:243]: so easy. Thank you :-)

File: Classes/Controller/OaiPmhController.php [hunk 2]
```diff
@@ -764,7 +776,7 @@ protected function generateResumptionTokenForDocumentListSet(array $documentList
 
         $resumptionTokenInfo = [];
         $resumptionTokenInfo['token'] = $resumptionToken;
-        $resumptionTokenInfo['cursor'] = $documentListSet['metadata']['completeListSize'] - count($documentListSet);
+        $resumptionTokenInfo['cursor'] = $currentCursor;
```
Comment (albig) [inline Classes/Controller/OaiPmhController.php:779]: This is correct for the `cursor` attribute following the linked docu. But we should change the XSLT file for the HTML-Output for this, don't we?

Kitodo Presentation  OAI 2 0 Request Results

Before your change, there was written "10" (for 10 documents per request).

File: Resources/Private/Templates/OaiPmh/Main.xml [hunk 3]
```diff
@@ -122,7 +124,9 @@
                         </f:else>
                     </f:if>
                 </f:for>
-                <resumptionToken cursor="{resumptionToken.cursor}" completeListSize="{resumptionToken.completeListSize}" expirationDate="<f:format.date format="Y-m-d\TH:i:s\Z">{resumptionToken.expired}</f:format.date>">{resumptionToken.token}</resumptionToken>
+                <f:if condition="{resumptionToken.token}">
```
Comment (albig) [inline Resources/Private/Templates/OaiPmh/Main.xml]: The `resumptionToken` is missing now on the last "page".

In the linked "FlowControl" it's written:

> The repository responds with an incomplete list of 75 records. The repository marks this list as the final incomplete list by including in the response an empty resumptionToken element with two attributes: a completeListSize of 175, and a cursor of 100.

**Review by albig — CHANGES_REQUESTED** [review]
Thank you for working on that. Please have a look at my comments.

File: Classes/Controller/OaiPmhController.php (hunk 1 above)
Comment (dvoracek-slub) [reply Classes/Controller/OaiPmhController.php:243]: :-)

File: Resources/Private/Templates/OaiPmh/Main.xml (hunk 3 above)
Comment (dvoracek-slub) [reply Resources/Private/Templates/OaiPmh/Main.xml]: Sorry for having missed that. Do you think the patch is right? As far as I understood, the `<resumptionToken>` tag should still be omitted if there is only one page, and there shouldn't be an `expirationDate` parameter on the last page. So I've extracted this to a separate section in the Fluid template.

File: Classes/Controller/OaiPmhController.php (hunk 2 above)
Comment (dvoracek-slub) [reply Classes/Controller/OaiPmhController.php:779]: I'm not sure what's expected here, but if it's just a more user-friendly representation of the XML file, perhaps we could just clarify that it's the "*Previously* Submitted Records"?

Comment (albig) [reply Classes/Controller/OaiPmhController.php:779]: Unfortunately, we do not have the amount of items in the current list. I've just looked around, how other solved this.

**Review by albig — APPROVED** [review]

**Pull request merged**